lib/improver/tests/nowcasting/forecasting/test_CreateExtrapolationForecast.py

```diff
@@ -0,0 +1,234 @@
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+""" Unit tests for the nowcasting.AdvectField plugin """
+
+import unittest
+import numpy as np
+
+from iris.tests import IrisTest
+
+from improver.nowcasting.forecasting import (
+    AdvectField, CreateExtrapolationForecast)
+from improver.tests.set_up_test_cubes import set_up_variable_cube
+from improver.tests.nowcasting.forecasting.test_AdvectField import (
+    set_up_xy_velocity_cube)
+
+
+def setup_orographic_enhancement_cube():
+    """Set up an orogrpahic enhancement cube using central utilities"""
+    data = np.array([[1, 1, 1],
+                     [0, 1, 0],
+                     [0, 0, 0],
+                     [0, 0, 0]], dtype=np.float32)
+    orographic_enhancement_cube = set_up_variable_cube(
+        data, name="orographic_enhancement", units="mm/hr",
+        spatial_grid="equalarea")
+    return orographic_enhancement_cube
+
+
+def setup_precipitation_cube():
+    """Set up an precipitation cube using central utilities"""
+    data = np.array([[1, 2, 1],
+                     [1, 1, 1],
+                     [0, 2, 0],
+                     [1, 2, 1]], dtype=np.float32)
+    precipitation_cube = set_up_variable_cube(
+        data, name="lwe_precipitation_rate", units="mm/hr",
+        spatial_grid="equalarea")
+    return precipitation_cube
+
+
+class SetUpCubes(IrisTest):
+    """
+    Set up a base class with a setUp method that can be inherited by all
+    the other unit test classes. This means they can all share the same
+    setUp method.
+    """
+    def setUp(self):
+        """Set up cubes needed for the __init__ method """
+        self.precip_cube = setup_precipitation_cube()
+        self.oe_cube = setup_orographic_enhancement_cube()
+        self.vel_x = set_up_xy_velocity_cube("advection_velocity_x")
+        self.vel_y = set_up_xy_velocity_cube("advection_velocity_y")
+        for cube in [self.precip_cube, self.oe_cube]:
+            cube.coord("projection_x_coordinate").points = 600*np.arange(3)
+            cube.coord("projection_y_coordinate").points = 600*np.arange(4)
+
+
+class Test__init__(SetUpCubes):
+    """Test class initialisation"""
+
+    def test_basic(self):
+        """Test for simple case where __init__ does not change the input."""
+        # Change the input cube so no orographic enhancement is expected.
+        input_cube = self.precip_cube.copy()
```

Comment:
It was only so I was certain that when I'm comparing the plugin.input_cube, it is definitely something different to the cube I passed in. We've had problems where plugins are modifying input cubes, rather than returning new cubes, so this was trying to make the behaviour clear.

In this particular case I was also trying to make clear that I'm not passing in a precip_cube, but something else.

I am happy to change them all if you want.

Reply:
In response to the first point, the unit-test always passes in input_cube.copy(), so this is still satisfied.
Yes, I agree that this isn't a precip cube once you've renamed it, so I suppose it does make it a bit clearer. I was just thinking that it was an unnecessary use of memory, but these are tiny unit-test cubes, so making more copies isn't a problem. I'm happy for you to leave these as they currently are.